backend/inspirehep/mailing/views.py

```diff
@@ -5,9 +5,31 @@
 # inspirehep is free software; you can redistribute it and/or modify it under
 # the terms of the MIT License; see LICENSE file for more details.
 
+import structlog
+from flask import Blueprint, jsonify, request
 
-from flask import Blueprint
+from .api.jobs import subscribe_to_jobs_weekly_list
 
 blueprint = Blueprint(
     "inspirehep_mailing", __name__, template_folder="templates", url_prefix=""
 )
+
+
+LOGGER = structlog.getLogger()
+
+
+@blueprint.route("mailing/jobs/weekly/subscribe", methods=["POST"])
+def subscribe_jobs_weekly():
+    try:
+        data = request.get_json()
+        subscribe_to_jobs_weekly_list(
+            data["email"], data["first_name"], data["last_name"]
+        )
+        LOGGER.info("User successfuly subscribed.", user=data)
+        return jsonify({"message": "Succesfully subscribed."}), 200
+    except ValueError:
+        LOGGER.info("Invalid email address.", user=data)
+        return jsonify({"message": "Not a valid email address."}), 400
+    except Exception:
+        LOGGER.exception("Cannot subscribe user to list.", user=data)
+        return jsonify({"message": "Unexpected error."}), 500
```

Comment:
Is it possible to get a `KeyError`? In that case the error message can be more helpful, like `Missing field email`

Reply:
@ammirate updated with Marshmallow validation